packages/wrangler/src/deprecated/index.ts

```diff
@@ -47,19 +41,22 @@ export function generateOptions(yargs: Argv<CommonYargsOptions>) {
 			deprecated: true,
 		});
 }
+type GenerateArgs = YargsOptionsToInterface<typeof generateOptions>;
 
-export async function generateHandler({
+//TODO: move this out of deprecated since apparently it's not deprecated anymore
+export async function generateHandler(args: GenerateArgs) {
 	// somehow, `init` marks name as required but then also runs fine
 	// with the name omitted, and then substitutes it at runtime with ""
-	name = "",
-	template,
-	type,
-	site,
-	...args
-}: ArgumentsCamelCase<GenerateArgs>) {
+	let { name = "", template } = args;
+	const { type, site } = args;
```

Comment:
I think that destructuring `args` here is not that helpful, and it might well lead to simpler code to just access the `args` properties as needed.
I note that despite pulling out `type` here, we also use `args.type` below.

Slightly off-topic, I still feel this bit of code to be pretty evil!

```ts
	if (isRemote(name)) {
		[template, name] = [name, template];
	}
```

I would love a refactor to remove that!

Reply:
💯 - v cursed